Failed to build with rustc 1.21.0-nightly (ed16b0a1d 2017-08-05)

Due to: rust-lang/rust#43554
Logs:
```
error[E0599]: no associated item named `F32` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:407:19
    |
407 |             val @ ConstFloat::F32(_) => {
    |                   ^^^^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F64` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:417:19
    |
417 |             val @ ConstFloat::F64(_) => {
    |                   ^^^^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F32` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:408:28
    |
408 |                 let zero = ConstFloat::F32(0.0);
    |                            ^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F32` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:410:32
    |
410 |                 let infinity = ConstFloat::F32(::std::f32::INFINITY);
    |                                ^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F32` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:412:36
    |
412 |                 let neg_infinity = ConstFloat::F32(::std::f32::NEG_INFINITY);
    |                                    ^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F64` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:418:28
    |
418 |                 let zero = ConstFloat::F64(0.0);
    |                            ^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F64` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:420:32
    |
420 |                 let infinity = ConstFloat::F64(::std::f64::INFINITY);
    |                                ^^^^^^^^^^^^^^^

error[E0599]: no associated item named `F64` found for type `rustc_const_math::ConstFloat` in the current scope
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.148/src/misc.rs:422:36
    |
422 |                 let neg_infinity = ConstFloat::F64(::std::f64::NEG_INFINITY);
    |                                    ^^^^^^^^^^^^^^^

error: aborting due to 8 previous errors

error: Could not compile `clippy_lints`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `clippy v0.0.148`, intermediate artifacts can be found at `/tmp/cargo-install.qq9jCb3lqqwF`

Caused by:
  build failed
```

--- a/clippy_lints/src/misc.rs
+++ b/clippy_lints/src/misc.rs
@@ -12,7 +12,7 @@ use utils::{get_item_name, get_parent_expr, implements_trait, in_macro, is_integ
             span_lint, span_lint_and_then, walk_ptrs_ty, last_path_segment, iter_input_pats, in_constant,
             match_trait_method, paths};
 use utils::sugg::Sugg;
-use syntax::ast::{LitKind, CRATE_NODE_ID};
+use syntax::ast::{LitKind, CRATE_NODE_ID, FloatTy};
 
 /// **What it does:** Checks for function arguments and let bindings denoted as `ref`.
 ///
@@ -403,23 +403,41 @@ fn is_allowed(cx: &LateContext, expr: &Expr) -> bool {
     let res = ConstContext::new(cx.tcx, cx.param_env.and(substs), cx.tables).eval(expr);
     if let Ok(ConstVal::Float(val)) = res {
         use std::cmp::Ordering;
-        match val {
-            val @ ConstFloat::F32(_) => {
-                let zero = ConstFloat::F32(0.0);
+        match val.ty {
+            FloatTy::F32 => {
+                let zero = ConstFloat {
+                    ty: FloatTy::F32,
+                    bits: 0.0f32.to_bits() as u128,
+                };
 
-                let infinity = ConstFloat::F32(::std::f32::INFINITY);
+                let infinity = ConstFloat {
+                    ty: FloatTy::F32,
+                    bits: ::std::f32::INFINITY.to_bits() as u128,
+                };
 
-                let neg_infinity = ConstFloat::F32(::std::f32::NEG_INFINITY);
+                let neg_infinity = ConstFloat {
+                    ty: FloatTy::F32,
+                    bits: ::std::f32::NEG_INFINITY.to_bits() as u128,
+                };
 
                 val.try_cmp(zero) == Ok(Ordering::Equal) || val.try_cmp(infinity) == Ok(Ordering::Equal) ||
                 val.try_cmp(neg_infinity) == Ok(Ordering::Equal)
             },
-            val @ ConstFloat::F64(_) => {
-                let zero = ConstFloat::F64(0.0);
+            FloatTy::F64 => {
+                let zero = ConstFloat {
+                    ty: FloatTy::F64,
+                    bits: 0.0f64.to_bits() as u128,
+                };
 
-                let infinity = ConstFloat::F64(::std::f64::INFINITY);
+                let infinity = ConstFloat {
+                    ty: FloatTy::F64,
+                    bits: ::std::f64::INFINITY.to_bits() as u128,
+                };
 
-                let neg_infinity = ConstFloat::F64(::std::f64::NEG_INFINITY);
+                let neg_infinity = ConstFloat {
+                    ty: FloatTy::F64,
+                    bits: ::std::f64::NEG_INFINITY.to_bits() as u128,
+                };
 
                 val.try_cmp(zero) == Ok(Ordering::Equal) || val.try_cmp(infinity) == Ok(Ordering::Equal) ||
                 val.try_cmp(neg_infinity) == Ok(Ordering::Equal)

